### PR TITLE
Fix bugs with RC

### PR DESCRIPTION
**Pull request by nathanstitt**
* The course nag wasn't updating when the course changed causing it to not show up when you navigated from a context without a course to one that had it
 * The terms modal also wasn't being passed the course on the new course wizard (because it didn't have one).  Renamed the prop to "canBeDisplayed" and used that.
* The course ux was loading tasks for the students whenever it was mount, but two instances were being opened one on the navbar, and one on student dashboard.  I first removed loading from the UX and added it to the student tasks, but then realized that it wasn't needed onthe student dashboard so I removed it from there altogether.

### CODE REVIEW

File: tutor/src/models/course/onboarding/student-course.js [hunk 1]
```diff
@@ -79,41 +76,19 @@ export default class StudentCourseOnboarding extends BaseOnboarding {
...
-      this.refreshTasksTimer = setTimeout(this.fetchTaskPeriodically, interval);
-    });
-  }
 
   mount() {
     super.mount();
-    if (!this.refreshTasksTimer && !this.paymentIsPastDue) {
-      this.fetchTaskPeriodically();
-    }
+//    this.course.studentTasks.startFetching();
```
Comment (TomWoodward) [inline tutor/src/models/course/onboarding/student-course.js]: commented code

File: tutor/src/models/course/onboarding/student-course.js [hunk 2]
```diff
@@ -79,41 +76,19 @@ export default class StudentCourseOnboarding extends BaseOnboarding {
...
     this.tourContext.otherModal = this;
   }
 
   close() {
     super.close();
-    if (this.refreshTasksTimer) {
-      clearInterval(this.refreshTasksTimer);
-      this.refreshTasksTimer = null;
-    }
+  //  this.course.studentTasks.stopFetching();
```
Comment (TomWoodward) [inline tutor/src/models/course/onboarding/student-course.js]: commented code

File: tutor/src/models/exercises/exercise.js [hunk 3]
```diff
@@ -28,7 +28,6 @@ export default class TutorExercise extends BaseModel {
 
   @hasMany({ model: Tag, inverseOf: 'exercise', extend: extendHasMany({
     importantInfo() {
-      //console.log("IMP", filter(this, 'isImportant'))
```
Comment (TomWoodward) [inline tutor/src/models/exercises/exercise.js]: commented code

File: tutor/src/api/index.coffee [hunk 4]
```diff
@@ -196,7 +196,7 @@ startAPI = ->
     pattern: 'enrollment/{id}/approve', method: 'PUT'
     onSuccess: 'onApiRequestComplete', onFail: 'setApiErrors')
 
-  connectModelRead(CourseStudentTasks, 'fetch', onSuccess: 'onLoaded', pattern: 'courses/{courseId}/dashboard')
+  connectModelRead(CourseStudentTasks, 'fetch', onSuccess: 'onLoaded', pattern: 'courses/{course.id}/dashboard')
```
Comment (TomWoodward) [inline tutor/src/api/index.coffee]: how is this url pattern used? is it the same router as here? https://github.com/openstax/tutor-js/pull/2378/files#diff-10c23c10ea6ef74f0e103f771283948eR107

i didn't see where you changed any route params from courseId to course

Comment (nathanstitt) [reply tutor/src/api/index.coffee]: Yeah, that change was a side effect of this: https://github.com/openstax/tutor-js/pull/2378/files#diff-27d249dcdd15872c8db5b5f36317bce4L22

StudentTasks used to just store the courseId, but it now stores the course.

**Comment by nathanstitt** [comment]
card: https://app.zenhub.com/workspace/o/openstax/tutor/issues/25

**Pull request merged**